src/components/image-plus.js

```diff
@@ -0,0 +1,74 @@
+AFRAME.registerComponent("image-plus", {
+  dependencies: ["geometry", "material"],
+
+  schema: {
+    src: { type: "string" },
+
+    initialOffset: { default: { x: 0, y: 0, z: -1.5 } },
+    reorientOnGrab: { default: false }
+  },
+
+  _fit(w, h) {
+    const ratio = (h || 1.0) / (w || 1.0);
+    const geo = this.el.geometry;
+    let width, height;
+    if (geo && geo.width) {
+      if (geo.height && ratio > 1) {
+        width = geo.width / ratio;
+      } else {
+        height = geo.height * ratio;
+      }
+    } else if (geo && geo.height) {
+      width = geo.width / ratio;
```

Comment:
Unclear to me what the assumptions are for geo.height and geo.width. Seems like if this line runs, geo.width will be 0 or undefined

Reply:
Ah yeah, this code is from https://github.com/chenzlabs/aframe-src-fit-component/blob/master/index.js never really went back and looked it over. The intent is that if the geometry has a fixed width and height, those are used as the bounds, otherwise if just one of them is defined, that is used as the bounds, otherwise 1 is used as the bounds.